demos/bubble_shear.py

```diff
@@ -0,0 +1,232 @@
+# Advection of a tracer bubble in a non-uniform shear flow
+# ===================================================
+
+# In this demo, we consider the advection of a scalar concentration field, similarly to
+# our previous solid body rotation demo. However, we now consider a background velocity
+# field that is not uniform in time. We further prescribe homogeneous Dirichlet boundary
+# conditions for the concentration, so in this demo we solve
+#
+# .. math::
+#   \begin{array}{rl}
+#       \frac{\partial c}{\partial t} + \nabla\cdot(\mathbf{u}c)=0& \text{in}\:\Omega\\
+#       c=0 & \text{on}\:\partial\Omega,\\
+#       c=c_0(x,y) & \text{at}\:t=0,
+#   \end{array},
+#
+# where :math:`c=c(x,y,t)` is the sought tracer concentration, :math:`\mathbf{u}=\mathbf{u}(x,y,t)` is the
+# background velocity field, and :math:`\Omega=[0, 1]^2` is the spatial domain of
+# interest.
+#
+# First, we import Firedrake and Goalie. ::
+
+from firedrake import *
+from goalie import *
+
+# We begin by definining the background velocity field :math:`\mathbf{u}(x, y, t)`.
+# Specifically, we choose the velocity field that is periodic in time, such as
+# in TODO: cite, and is given by
```

Comment:
I added the citation in demo_references.bib but not sure how to make the citation like in other demos.